plugins/LadspaEffect/caps/CMakeLists.txt

```diff
@@ -27,12 +27,3 @@ IF(LMMS_BUILD_WIN32)
 		COMMAND_EXPAND_LISTS
 	)
 ENDIF(LMMS_BUILD_WIN32)
-
-IF(NOT LMMS_BUILD_APPLE AND NOT LMMS_BUILD_OPENBSD)
-	SET_TARGET_PROPERTIES(caps PROPERTIES LINK_FLAGS "${LINK_FLAGS} -shared -Wl,-no-undefined")
```

Comment:
The entire if condition looks sus, from my logic, only linux seems to use that flag properly but in the next line, for linux, everything seems to get reset. I will add it back in tho.

Reply:
That's a good point - I'll see if I can figure out why it was done like this.

Reply:
Update?

Reply:
It looks like this was a mistake in the initial conversion from Automake to CMake sixteen years ago. I think it makes sense to remove it for now, given that these flags have been overwritten so far.

`-shared` is another one of the flags that we shouldn't specify manually, and should instead leave to CMake to pass based on the target type. `-Wl,-no-undefined`, on the other hand, is potentially useful for catching errors in Linux builds. The same errors will be caught by macOS and Windows builds for cross-platform code, but it would still be beneficial for Linux-specific code, or for developers using Linux who can have errors reported locally, rather than via CI build logs. However, it seems a bit odd to pass it here alone, and we can add it elsewhere at a later date.

Reply:
So basically, looks like it's fine to remove but `-Wl,no-undefined` seems to have some use, which can be added later if we should.